### PR TITLE
feat(ingest/snowflake): allow shares config without platform instance

### PR DESCRIPTION
**Pull request by mayurinehate**
## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [x] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/tests/unit/test_snowflake_shares.py [hunk 1]
```diff
@@ -336,13 +338,85 @@ def test_snowflake_shares_workunit_inbound_and_outbound_share(
...
                 ]
+
+
+def test_snowflake_shares_workunit_inbound_and_outbound_share_no_platform_instance(
+    snowflake_databases: List[SnowflakeDatabase],
+) -> None:
+    config = SnowflakeV2Config(
+        account_id="abc12345",
+        shares={
+            "share1": SnowflakeShareConfig(
```
Comment (mayurinehate) [inline metadata-ingestion/tests/unit/test_snowflake_shares.py:361]: "db1" (primary, included in share) and "db2"(secondary, created from share) are present in this account, as present in `snowflake_databases`.

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_config.py [hunk 2]
```diff
@@ -247,10 +247,10 @@ def validate_shares(
 
         if shares:
             # Check: platform_instance should be present
-            assert current_platform_instance is not None, (
-                "Did you forget to set `platform_instance` for current ingestion ? "
-                "It is required to use `platform_instance` when ingesting from multiple snowflake accounts."
-            )
+            if current_platform_instance is None:
+                logger.warning(
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_config.py:251]: I'd make this an INFO, and perhaps explain what setting platform instance provides (i.e. no conflict for entities with the same name in each instance). I think there are valid use cases for not setting platform instance, e.g. you don't expect overlap or you're looking at different slices of the same data

**Review by asikowitz — APPROVED** [review]

**Comment by asikowitz** [comment]
Looks like you have a 3.7 lint failure -- missing type annotation. I'm confused how this passed 3.10

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_config.py (hunk 2 above)
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_config.py]: done

**Pull request merged**